src/query/impl/RTreeWrapper.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <map>
+#include "RTreeAdapter.hpp"
+
+namespace precice {
+
+namespace testing {
+namespace accessors {
+struct rtree;
+}
+} // namespace testing
+
+namespace query {
+namespace impl {
+
+using VertexTraits   = impl::RTreeTraits<mesh::Vertex>;
+using EdgeTraits     = impl::RTreeTraits<mesh::Edge>;
+using TriangleTraits = impl::RTreeTraits<mesh::Triangle>;
+
+struct MeshIndices {
+  VertexTraits::Ptr   vertexRTree;
+  EdgeTraits::Ptr     edgeRTree;
+  TriangleTraits::Ptr triangleRTree;
+};
+
+/// Class to encapsulate boost::geometry implementations
+class RTreeWrapper {
+public:
+  /// Return vertex index tree from cache, if cache is empty, create the tree
+  VertexTraits::Ptr getVertexRTree(const mesh::PtrMesh &mesh);
+
+  /// Return edge index tree from cache, if cache is empty, create the tree
+  EdgeTraits::Ptr getEdgeRTree(const mesh::PtrMesh &mesh);
+
+  /// Return triangle index tree from cache, if cache is empty, create the tree
+  TriangleTraits::Ptr getTriangleRTree(const mesh::PtrMesh &mesh);
+
+  /// Return boost::geometry version of enclosing box around a point
+  Box3d getEnclosingBox(const mesh::Vertex &middlePoint, double sphereRadius);
```

Comment:
This doesn't belong here.
Maybe this could go into a utility file of the mesh?

Reply:
If I move this i need to include `boost::geometry` there too. Does it worth it?

Reply:
Just realised that you made this method obsolete by implementing `Index::getVerticesInsideBox()` :wink: